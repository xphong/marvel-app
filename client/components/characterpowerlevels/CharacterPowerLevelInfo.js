import PropTypes from 'prop-types';
import React, { Component } from 'react';

import Character from '../characters/Character';
import CharacterPowerLevelRadarChart from '../characterpowerlevels/CharacterPowerLevelRadarChart';
import CharacterPowerLevelBarChart from '../characterpowerlevels/CharacterPowerLevelBarChart';

export default class CharacterPowerLevelInfo extends Component {
  getCharacterChartData(character) {
    const labels = ['Intelligence', 'Strength', 'Speed', 'Durability', 'Energy Projection', 'Fighting Ability'];
    const label = 'Character Data Set';
    const backgroundColor = ['#FF6384', '#4BC0C0', '#FFCE56', '#E7E9ED', '#36A2EB'];
    const data = [
      character.Intelligence,
      character.Strength,
      character.Speed,
      character.Durability,
      character.Energy_Projection,
      character.Fighting_Ability
    ];

    const characterChartData = {
      labels,
      datasets: [
        {
          label,
          backgroundColor,
          data
        }
      ]
    };

    return characterChartData;
  }

  render() {
    const { data } = this.props.powerLevels;
    const { id } = this.props;
    const character = data[id];

    if (!character) return null;

    const characterChartData = this.getCharacterChartData(character);

    return (
      <div className="ui container center aligned">
        <Character character={character} />
        <div className="chart-area">
          <CharacterPowerLevelRadarChart characterChartData={characterChartData} />
          <CharacterPowerLevelBarChart characterChartData={characterChartData} />
        </div>
      </div>
    );
  }
}

CharacterPowerLevelInfo.propTypes = {
  powerLevels: PropTypes.object.isRequired
};
