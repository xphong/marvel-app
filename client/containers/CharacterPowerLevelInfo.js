import PropTypes from 'prop-types';
import React, { Component } from 'react';

import CharacterPowerLevelInfoDisplay from '../components/characterpowerlevels/CharacterPowerLevelInfoDisplay';
import CharacterPowerLevelRadarChart from '../components/characterpowerlevels/CharacterPowerLevelRadarChart';
import CharacterPowerLevelBarChart from '../components/characterpowerlevels/CharacterPowerLevelBarChart';

export default class CharacterPowerLevelInfo extends Component {
  constructor(props){
    super(props);
  }

  getCharacterChartData(character) {
    const labels = ['Intelligence', 'Strength', 'Speed', 'Durability', 'Energy Projection', 'Fighting Ability'];
    const label = 'Character Data Set';
    const backgroundColor = ['#FF6384', '#4BC0C0', '#FFCE56', '#E7E9ED', '#36A2EB'];
    const data = [];

    data.push(character.Intelligence);
    data.push(character.Strength);
    data.push(character.Speed);
    data.push(character.Durability);
    data.push(character.Energy_Projection);
    data.push(character.Fighting_Ability);

    const characterChartData = {
      labels,
      datasets: [
        {
          label,
          backgroundColor,
          data
        }
      ]
    };

    return characterChartData;
  }

  render() {
    const {data} = this.props.powerLevels;
    const {id} = this.props;
    const character = data[id];

    if (!character) return null;

    const characterChartData = this.getCharacterChartData(character);

    return (
      <div className='ui container center aligned'>
        <CharacterPowerLevelInfoDisplay character={character} />
        <div className='chart-area'>
          <CharacterPowerLevelRadarChart characterChartData={characterChartData} />
          <CharacterPowerLevelBarChart characterChartData={characterChartData} />
        </div>
      </div>
    );
  }
}

CharacterPowerLevelInfo.propTypes = {
  powerLevels: PropTypes.object.isRequired
};
