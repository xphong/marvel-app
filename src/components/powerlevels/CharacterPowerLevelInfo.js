import React, { Component, PropTypes } from 'react';

import CharacterPowerLevelRadarChart from './CharacterPowerLevelRadarChart';

export default class CharacterPowerLevelInfo extends Component {
  constructor(props){
    super(props);
  }

  render() {
    const {data} = this.props.powerLevels;
    const {id} = this.props;
    const currentCharacter = data[id];

    if (!currentCharacter) {
      return null;
    }

    return (
      <div>
        <h1>{currentCharacter.Name}</h1>
        <a target='_blank' href={currentCharacter.Profile_Link}>
          <img className='ui medium image transition' src={currentCharacter.Image_Link} />
        </a>
        <h4>Average Power Level: {currentCharacter.AveragePowerLevel}</h4>
        <CharacterPowerLevelRadarChart character={currentCharacter} />
      </div>
    );
  }
}

CharacterPowerLevelInfo.propTypes = {
  powerLevels: PropTypes.object.isRequired
};
