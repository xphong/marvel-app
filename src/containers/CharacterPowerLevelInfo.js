import React, { Component, PropTypes } from 'react';

import CharacterPowerLevelInfoDisplay from '../components/characterpowerlevels/CharacterPowerLevelInfoDisplay';
import CharacterPowerLevelRadarChart from '../components/characterpowerlevels/CharacterPowerLevelRadarChart';

export default class CharacterPowerLevelInfo extends Component {
  constructor(props){
    super(props);
  }

  getCharacterChartData(character) {
    return {
      labels: ["Eating", "Drinking", "Sleeping", "Designing", "Coding", "Cycling", "Running"],
      datasets: [
        {
          label: "Character Data Set",
          data: [28, 48, 40, 19, 96, 27, 100]
        }
      ]
    };
  }

  render() {
    const {data} = this.props.powerLevels;
    const {id} = this.props;
    const character = data[id];

    if (!character) {
      return null;
    }

    const characterChartData = this.getCharacterChartData(character);

    return (
      <div className='ui container center aligned'>
        <CharacterPowerLevelInfoDisplay character={character} />
        <CharacterPowerLevelRadarChart characterChartData={characterChartData} />
      </div>
    );
  }
}

CharacterPowerLevelInfo.propTypes = {
  powerLevels: PropTypes.object.isRequired
};
