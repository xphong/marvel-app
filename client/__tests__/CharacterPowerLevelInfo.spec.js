import React from 'react';
import ReactDOM from 'react-dom';

import CharacterPowerLevelInfo from '../containers/CharacterPowerLevelInfo';

function setup({ isLoading = false, id = 0 } = {}) {
  const props = {
    actions: {
      fetchPowerLevels: jest.fn()
    },
    powerLevels: {
      isLoading,
      data: [
        {
          Name: 'Ant-Man (Hank Pym)',
          Profile_Link: 'http://marvel.wikia.com/Henry_Pym_(Earth-616)',
          Image_Link: 'http://i.annihil.us/u/prod/marvel/i/mg/9/c0/53176aa9df48d.jpg',
          Intelligence: '6',
          Strength: '7',
          Speed: '3',
          Durability: '3',
          Energy_Projection: '6',
          Fighting_Ability: '3'
        }
      ]
    },
    id
  };

  const wrapper = shallow(<CharacterPowerLevelInfo {...props} />);

  return {
    props,
    wrapper
  };
}

describe('CharacterPowerLevelInfo Container', () => {
  it('should render CharacterPowerLevelInfo', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
