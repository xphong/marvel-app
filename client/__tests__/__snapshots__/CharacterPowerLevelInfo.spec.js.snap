// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterPowerLevelInfo Container should render CharacterPowerLevelInfo 1`] = `
<div
  className="ui container center aligned"
>
  <Character
    character={
      Object {
        "Durability": "3",
        "Energy_Projection": "6",
        "Fighting_Ability": "3",
        "Image_Link": "http://i.annihil.us/u/prod/marvel/i/mg/9/c0/53176aa9df48d.jpg",
        "Intelligence": "6",
        "Name": "Ant-Man (Hank Pym)",
        "Profile_Link": "http://marvel.wikia.com/Henry_Pym_(Earth-616)",
        "Speed": "3",
        "Strength": "7",
      }
    }
  />
  <div
    className="chart-area"
  >
    <CharacterPowerLevelRadarChart
      characterChartData={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": Array [
                "#FF6384",
                "#4BC0C0",
                "#FFCE56",
                "#E7E9ED",
                "#36A2EB",
              ],
              "data": Array [
                "6",
                "7",
                "3",
                "3",
                "6",
                "3",
              ],
              "label": "Character Data Set",
            },
          ],
          "labels": Array [
            "Intelligence",
            "Strength",
            "Speed",
            "Durability",
            "Energy Projection",
            "Fighting Ability",
          ],
        }
      }
    />
    <CharacterPowerLevelBarChart
      characterChartData={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": Array [
                "#FF6384",
                "#4BC0C0",
                "#FFCE56",
                "#E7E9ED",
                "#36A2EB",
              ],
              "data": Array [
                "6",
                "7",
                "3",
                "3",
                "6",
                "3",
              ],
              "label": "Character Data Set",
            },
          ],
          "labels": Array [
            "Intelligence",
            "Strength",
            "Speed",
            "Durability",
            "Energy Projection",
            "Fighting Ability",
          ],
        }
      }
    />
  </div>
</div>
`;
